### PR TITLE
Add GDP share constraint for tech cost

### PR DESCRIPTION
**Pull request by jansteinhauser**
Change tech cost upper bound to a config-set GDP share

## :bird: Purpose of this PR :bird:

- This PR changes the tech cost upper bound in the endo TC realization to a permille share ofthe regional GDP (PPP). As the current default is purely technical, this is a necessary constraint for price-driven runs that may push TC and TC cost to extreme heights.

## :wrench: Checklist for PR creator :wrench:

- [x] Labeling pull request correctly [from the label list](https://github.com/magpiemodel/magpie/labels).

* Low risk : Simple bugfixes (missing files, updated documentation, typos) or Start/output scripts
* **Medium risk : New realization / Changes to existing realization / Other changes which don't modify default.cfg**
* High risk : New input files (if cfg$input is changed in default.cfg) / Modification to core model (eg. changes in equations, calculations, introduction of new sets etc.) / Other changes in default.cfg

- [x] Providing additional information based on PR label

* Low risk : No new model run needed.
* Medium risk : Default run based on the current version of the fork from which PR is made
* High risk
  * Default run from the current develop branch
  * Default run based on the current version of the fork from which PR is made

Costs_TC

Productivity_Landuse_Intensity_Indicator_Tau_cropland

- [x] :chart_with_downwards_trend: Performance loss/gain from current default behavior :chart_with_upwards_trend:
  * Current develop branch default : 33 mins
  * This PR's default :  37 mins

- [x] Added changes to `CHANGELOG.md`
- [x] Compilation check (model starts without compilation errors - use `gams main.gms action=c` in model folder for testing).
- [x] No hard coded numbers and cluster/country/region names.
- [x] The new code doesn't contain declared but unused parameters or variables.
- [x] Where relevant, In-code comments added including documentation comments.
- [x] Made sure that documentation created with [`goxygen`](https://github.com/pik-piam/goxygen) is okay (use `goxygen::goxygen()` for testing).
- [x] Changes to [`magpie4`](https://github.com/pik-piam/magpie4) R library for post processing of model output (ideally backward compatible).
- [x] Self-review of my own code.
- [x] For high risk runs: validation of major model indicators - Land-use, emissions, food prices, Tau.  %Delete this line in case it is not a high risk run%

## :warning: Additional notes or warnings :warning:
NA

## :rotating_light: Checklist for RSE reviewer :rotating_light:

- [x] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] No unnecessary increase in module interfaces
- [ ] All required updates in interfaces (if any) have been properly adressed in the module contracts
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

## :rotating_light: Checklist for MAgPIE reviewer :rotating_light:

- [x] PR is labeled correctly.
- [x] `CHANGELOG` is updated correctly
- [x] No hard coded numbers and cluster/country/region names.
- [x] Changes to the model are scientifically sound
- [x] In-code comments and documentation comments are satisfactory.
- [x] model behavior/performance is satisfactory.
- [x] Requested changes (if any) were applied correctly

### CODE REVIEW

**Comment by jansteinhauser** [comment]
It seems I can never request reviews nor label the pull request (if I had to guess, it's because I am a first-time contributor), so if someone would be so kind to label this PR as medium risk (as it's changing a realization) and enhancement, that would be grand!
Also @tscheypidi @flohump would you kindly review this PR?
Thank you!

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### changed
+ - **config** added s13_tech_cost_gdp_share setting for tech cost upper bound as share of GDP PPP
```
Comment (tscheypidi) [inline CHANGELOG.md:12]: The name of the scalar is a bit misleading. Would be good to make more clear that we are talking about an upper bound here. Perhaps you can write something like `s13_max_gdp_shr` instead.

File: modules/13_tc/endo_jan22/presolve.gms [hunk 2]
```diff
@@ -26,4 +26,10 @@ else
...
 
 vm_tau.up(h,tautype) = 2*pcm_tau(h,tautype);
-vm_tech_cost.up(i) = 10e9;
+
+* constrain tech cost either to 10e9 mio. USD05PPP per yr per region or to a
+* share of regional GDP
+vm_tech_cost.up(i) =
+	sum((i_to_iso(i,iso),ct), im_gdp_pc_ppp_iso(ct,iso) * im_pop_iso(ct,iso)) * s13_tech_cost_gdp_share;
+
+vm_tech_cost.L(i) = vm_tech_cost.up(i);
```
Comment (tscheypidi) [inline modules/13_tc/endo_jan22/presolve.gms:35]: Can you add a comment here explaining why the level is being used (and highlight that it is in fact necessary to help the solver finding a proper solution).

In addition: is that normally a capital "L" for level? I always thought it is a small "l".

**Review by tscheypidi — CHANGES_REQUESTED** [review]
Thanks for your first PR! (hooray)

Just 2 minor requests from my side.

File: modules/13_tc/endo_jan22/presolve.gms (hunk 2 above)
Comment (jansteinhauser) [reply modules/13_tc/endo_jan22/presolve.gms]: Added and changed both.
The capital L was just me being inconsistent.

File: CHANGELOG.md (hunk 1 above)
Comment (jansteinhauser) [reply CHANGELOG.md]: Done

**Review by tscheypidi — APPROVED** [review]
looks good

**Review by flohump — APPROVED** [review]
Looks good!

**Pull request merged**